app/routes/_gcn.circulars._archive._index/SortSelectorButton.tsx

```diff
@@ -0,0 +1,109 @@
+/*!
+ * Copyright © 2023 United States Government as represented by the
+ * Administrator of the National Aeronautics and Space Administration.
+ * All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { useSubmit } from '@remix-run/react'
+import {
+  Button,
+  ButtonGroup,
+  CardBody,
+  Grid,
+  Icon,
+  Radio,
+} from '@trussworks/react-uswds'
+import classNames from 'classnames'
+import type { ChangeEvent } from 'react'
+import { useState } from 'react'
+
+import DetailsDropdownContent from '~/components/DetailsDropdownContent'
+
+const sortOptions = { circularID: 'Date', relevance: 'Relevance' }
+
+function SortButton({
+  sort,
+  expanded,
+  ...props
+}: {
+  sort?: string
+  expanded?: boolean
+} & Omit<Parameters<typeof ButtonGroup>[0], 'segmented' | 'children'>) {
+  const slimClasses = 'height-4 padding-y-0'
+
+  return (
+    <ButtonGroup type="segmented" {...props}>
+      <Button type="button" className={`${slimClasses} padding-x-2`}>
+        Sorted By{' '}
+        {sortOptions[(sort as keyof typeof sortOptions) || 'circularID']}
```

Comment:
`sort` is unsanitized input from the user (it's from the URL query string), so you cannot assume that `sort` is one of the keys of `sortOptions`.